text/1099-renderComponent.md

```diff
@@ -0,0 +1,212 @@
+ */
+export function renderComponent(
+    /**
+     * The component definition to render.
+     *
+     * Any component that has had its manager registered is valid.
+     * For the component-types that ship with ember, manager registration 
+     * does not need to be worried about. 
+     */
+    component: object,
+    options: {
+        /**
+         * The element to render the component in to.
+         */
+        into: IntoTarget;
+
+        /**
+         * Optional owner. Defaults to `{}`, can be any object, but will need to implement the [Owner](https://api.emberjs.com/ember/release/classes/Owner) API for components within this render tree to access services.
+         */
+        owner?: object;
+
+        /**
+         * Configure the `document` and `isInteractive`
+         */
+        env?: { 
+            /**
+             * Defaults to globalThis.document.
+             */
+            document?: SimpleDocument | Document; 
+
+            /**
+             * When false, modifiers will not run.
+             */
+            isInteractive?: boolean; 
+        };
+
+        /**
+         * These args get passed to the rendered component
+         */
+        args?: Record<string, unknown>;
+  }
+): RenderResult | undefined {
+    /* ... implementation details ... */
+}
+```
+
+`RenderResult` is a subset of the currently a pre-existing `interface` from `@glimmer/interfaces`, but would be exposed as public API via the only the return type of `renderComponent`.
+
+It's shape is:
+```ts
+export interface RenderResult {
+    /**
+     * Destroys the render tree and removes all rendered content from the element rendered into.
+     */
+    destroy(): void
+
+    /**
+     * Re-renders the component
+     */
+    rerender(options?: { alwaysRevalidate: false }): void;
+}
+```
+
+When called, updates will be scheduled and fully interactive components will work as folks would intuitively expect.
+The details of _how_ are out of scope for this RFC, and could change as we make improvements to rendering.
+
+### The Parameters
+
+#### `owner` (defaults to `{}`)
+
+some object to represent the `Owner` (if not the host-app's `Owner`).
+
+#### `env.document` (default's to globalThis.document)
+
+The browser's `document`, for accessing needed APIs for rendering and managing DOM
+
+#### `env.isInteractive` (defaults to true)
+
+When isInteractive is false, modifiers don't run, when true, modifiers do run.
+
+#### `into`
+
+The element to render the compnoent into.
+
+#### `args`
+
+The args to pass to the component
+
+### Usage
+
+via modifier:
+```gjs
+import { modifier } from 'ember-modifier';
+import { destroy } from '@ember/destroyable';
+import { renderComponent from '@ember/renderer';
+
+const Demo = <template>hi</template>;
+const render = modifier((element) => {
+    let result = renderComponent(Demo, {
+        owner,
+        env: { document: document, isInteractive: true },
```

Comment:
for non-interactive envs likely we may have different function like `renderSSR`, so, wondering if we really need this flag

Reply:
great question -- I'd need to dig in to glimmer-vm to see what all isInteractive is used for.

I don't know how much benefit there is to having renderComponent do multiple things via this flag vs your suggestion of renderSSR (which I like)